### PR TITLE
Refactor BIRReader into smaller files

### LINKED ISSUE #45: Count attribute values need to be checked
**Description:**
In a few places where `count` attributes (e.g. `importCount`, `constCount` ) are being parsed, the read value is assumed to be zero and the next bits are read for parsing. A non-zero check needs to be done and if non-zero `readS4be()` needs to be called appropriately to advance the reader. 
https://github.com/ballerina-platform/nballerina/blob/b2b985d473ac589aae90c53e92939aba651ccf92/BIR2llvmIR/BIRReader.cpp#L950

**Suggested Labels:**
Bug, Low priority

### LINKED ISSUE #158: Refactor BIRReader.h/.cpp into smaller files
**Description:**

Refactor BIRReader.h and cpp files into multiple smaller files similar to what #114 did for BIR.h.
These new files can be moved in to a sub directory inside the compiler folder.

**Describe your problem(s)**

**Describe your solution(s)**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by ruvi-d**
## Purpose

Description in #158  

Resolves #158 
Resolves #45 

## Approach

Created a new `reader` subfolder inside `compiler` and separated the BIRReader files into:
* BIRFileReader
* BIRReadBasicBlock
* BIRReadFunction
* BIRReadInsn
* BIRReadPackage
* ConstantPool

### CODE REVIEW

File: compiler/reader/BIRReadInsns.cpp [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+        currentBB.addNonTermInsn(
+            std::make_unique<ConstantLoadInsn>(std::move(lhsOp), currentBB, cp.getStringCp(valueCpIndex)));
+        return;
+    }
+    case TYPE_TAG_NIL: {
+        currentBB.addNonTermInsn(std::make_unique<ConstantLoadInsn>(std::move(lhsOp), currentBB));
+        return;
+    }
+    default: {
+        // add an error msg
```
Comment (shubhamnarlawar) [inline compiler/reader/BIRReadInsns.cpp:241]: Do you mean to add a fprint("Invalid TypeTag") statement here

**Comment by shubhamnarlawar** [comment]
@KavinduZoysa - Any suggestions on this PR

File: compiler/reader/BIRReadInsns.cpp (hunk 1 above)
Comment (ruvi-d) [reply compiler/reader/BIRReadInsns.cpp]: Yes.. Pushed commit with added print statements.

**Review by shubhamnarlawar — APPROVED** [review]
Changes looks fine to me.
Good to merge.

File: compiler/reader/ConstantPool.cpp [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+        int32_t valueSpaceSize = reader.readS4be();
+        for (auto i = 0; i < valueSpaceSize; i++) {
+            [[maybe_unused]] int32_t fTypeCpIndex = reader.readS4be();
+            int32_t valueLength = reader.readS4be();
+            for (auto j = 0; j < valueLength; j++) {
+                [[maybe_unused]] int8_t tmp = reader.readU1();
+            }
+        }
+
+        break;
```
Comment (KavinduZoysa) [inline compiler/reader/ConstantPool.cpp:308]: extra line.

Comment (shubhamnarlawar) [reply compiler/reader/ConstantPool.cpp:307]: I think extra line is not getting removed by clang-format.

**Review by KavinduZoysa — APPROVED** [review]

**Pull request merged**